### PR TITLE
feat: YAML parser to read the checklist items

### LINKED ISSUE #55: Write a YAML parser to read the checklist items into a Python object

### PR DESCRIPTION
**Pull request by SoloSynth1**
This PR adds a YAML parser for reading the checklist into a Python object.

Example codes have also been written to illustrate how to use the newly created class to obtain the tests.

Since the codes use `ruamel.yaml`, the conda environment file has been updated to include this additional dependency.

Closes #55.

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -4,211 +4,14 @@ channels:
...
-  - yarl=1.9.4=py312he37b823_0
-  - zeromq=4.3.5=h5119023_3
-  - zipp=3.17.0=pyhd8ed1ab_0
+  - jupyterlab=4.0
+  - pip=24.0
+  - python=3.12
+  - python-dotenv=1.0.1
+  - fire=0.6.0
+  - langchain=0.1.16
+  - ruamel.yaml=0.18.6
```
Comment (tonyshumlh) [inline environment.yml:14]: Should also include 'fire'

File: src/checklist/checklist.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import os
+from typing import Union
+
+import fire
+from ruamel.yaml import YAML
+
+
+class Checklist:
```
Comment (tonyshumlh) [inline src/checklist/checklist.py:8]: recommend adding function 'get_all_tests' and 'get_test_areas'

**Review by tonyshumlh — COMMENTED** [review]
Tested. It works except for missing the 'fire' package in environment.yml

File: environment.yml (hunk 1 above)
Comment (SoloSynth1) [reply environment.yml:14]: fire is already in the list (line 12)

**Comment by SoloSynth1** [comment]
> Tested. It works except for missing the 'fire' package in environment.yml

`fire` is already listed as a dependency in `environment.yml`. Refer to comment above.

File: src/checklist/checklist.py (hunk 2 above)
Comment (SoloSynth1) [reply src/checklist/checklist.py:8]: Added the above methods in the latest commit.

**Comment by tonyshumlh** [comment]
My bad. I used the old environment.
Tested. It works now.

**Comment by SoloSynth1** [comment]
Thanks @tonyshumlh, I assume this means this PR is ready to be merged?

**Review by tonyshumlh — APPROVED** [review]

**Pull request merged**